.github/workflows/php-hook-documentation.yml

```diff
@@ -0,0 +1,57 @@
+name: PHP Hook Documentation Generator
+
+on:
+  push:
+    branches:
+      - "release/**"
+    paths:
+      - "**.php"
+      - .github/workflows/php-hook-documentation.yml
+  pull_request:
+    types:
+      - opened
+    branches:
+      - "release/**"
+    paths:
+      - "**.php"
+      - .github/workflows/php-hook-documentation.yml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  HookDocumentation:
+    name: Hook Documentation Generator
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          # Checks out a branch instead of a commit in detached HEAD state
+          ref: ${{ github.head_ref }}
+
+        # This generates the documentation string. The `id` property is used to reference the output in the next step.
+      - name: Generate hook documentation
+        id: generate-hook-docs
+        uses: woocommerce/grow/hook-documentation@actions-v1
+        with:
+          debug-output: yes
+          source-directories: src/,views/
```

Comment:
What about the main plugin file `google-listings-and-ads.php`? Is this strictly a list of directories or does it support single files as well?

It so happens we don't have any hooks there now, but considering we want to use it for other extensions the script should be flexible enough to include the main plugin file. If that's not possible then can we log an issue in the Grow repo that this should be fixed?

Reply:
Seems like its possible as internally it uses Finder()->path()

You can see it here in a test when I added `gla_test_action` inside the main plugin file.

<img width="1328" alt="Screenshot 2023-08-18 at 17 03 01" src="https://github.com/woocommerce/google-listings-and-ads/assets/5908855/696a68ef-7675-4b24-80bd-23b7696059e2">

https://github.com/woocommerce/google-listings-and-ads/actions/runs/5902917550/job/16011805311